lib/shared/puppet_forge/module_name.rb

```diff
@@ -0,0 +1,23 @@
+module PuppetForge
+  # Normalize Forge module names
+  #
+  # Forge modules use '-' as the separator between the username and modulename
+  # in the module slug, but the Puppet Forge uses '/' as the separator on module
+  # pages and in the URL. This class hides the distinction by always using '-' as
+  # the separator for normal use. If the name needs to use the slash separated name
+  # then the #forge_name method can be used.
+  #
+  # @api private
+  class ModuleName < String
+
+    def initialize(*args)
+      super
+      tr!('/', '-')
+    end
+
+    # @return [String] The slash separated module name.
+    def forge_name
+      String.new(self).tr('-', '/')
+    end
```

Comment:
That's what I originally did, but if we do `self.tr` then we'll create another ModuleName object (I think) that will normalize it back the other way. I really need to write tests to verify this though.

I think we should have this method provided in some fashion, but I think that `#to_path` is a better name.

But in general, what do you think of the approach of wrapping interactions with module names in this object to remove the `#tr` calls all over the place?